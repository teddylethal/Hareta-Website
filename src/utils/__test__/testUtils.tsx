import { screen, type waitForOptions } from '@testing-library/dom'
import { render } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import App from 'src/App'
import { BrowserRouter } from 'react-router-dom'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import React from 'react'

export const delay = (time: number) =>
  new Promise((resolve) => {
    setTimeout(() => {
      resolve(true)
    }, time)
  })

export const logScreen = async (
  body: HTMLElement = document.body.parentElement as HTMLElement,
  options?: waitForOptions
) => {
  const { timeout = 1000 } = options || {}
  await delay(timeout)

  screen.debug(body, 999999999)
}

const createWrapper = () => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false
      },
      mutations: {
        retry: false
      }
    }
  })

  const Provider = ({ children }: { children: React.ReactNode }) => (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  )
  return Provider
}

const Provider = createWrapper()

export const renderWithRouter = ({ route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route)
  return {
    user: userEvent.setup(),
    ...render(
      <Provider>
        <App />
      </Provider>,
      { wrapper: BrowserRouter }
    )
  }
}
