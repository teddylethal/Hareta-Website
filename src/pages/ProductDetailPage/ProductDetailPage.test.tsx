import { delay, renderWithRouter } from 'src/utils/__test__/testUtils'
import { describe, expect, test } from 'vitest'

describe('ProductDetail', () => {
  test('Render UI product detail', async () => {
    renderWithRouter({ route: '/store/Koenigsegg-Gemera-i:3mPHrsiPZbRLaq' })
    await delay(5000)
    expect(document.body).toMatchSnapshot()
  })
})
