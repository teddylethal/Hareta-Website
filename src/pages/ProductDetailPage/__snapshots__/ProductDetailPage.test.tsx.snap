// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductDetail > Render UI product detail 1`] = `
<body
  data-theme="dark"
>
  <div>
    <div
      class="dark"
      style="min-height: inherit;"
    >
      <div
        class="flex h-full min-h-full shrink-0 flex-col justify-between bg-lightBg text-darkText duration-200 dark:bg-darkBg dark:text-lightText"
        style="min-height: inherit;"
      >
        <div
          class="fixed z-20 w-full"
        >
          <div
            style="transform: none;"
          >
            <div
              class="top-0 flex h-10 w-full items-center bg-lightHeader shadow-md duration-200 dark:bg-darkHeader tablet:h-12 desktop:h-16"
            >
              <div
                class="container grid w-full grid-cols-3 items-center text-black duration-200  dark:text-white"
              >
                <div
                  class="col-span-2 flex items-center justify-start space-x-2 text-sm font-medium uppercase tablet:font-semibold tabletLarge:text-base desktop:space-x-4 desktop:text-lg"
                >
                  <a
                    class="rounded-md border border-none p-1 hover:text-primaryColor dark:hover:text-primaryColor"
                    href="/"
                  >
                    <p>
                      Home
                    </p>
                  </a>
                  <a
                    aria-current="page"
                    class="rounded-md border border-none p-1 hover:text-primaryColor dark:hover:text-primaryColor text-primaryColor dark:text-primaryColor"
                    href="/store"
                  >
                    <p>
                      Store
                    </p>
                  </a>
                  <a
                    class="rounded-md border border-none p-1 hover:text-primaryColor dark:hover:text-primaryColor"
                    href="/events"
                  >
                    <p>
                      Event
                    </p>
                  </a>
                  <a
                    class="rounded-md border border-none p-1 hover:text-primaryColor dark:hover:text-primaryColor"
                    href="/blogs"
                  >
                    <p>
                      Blog
                    </p>
                  </a>
                  <div
                    class="group"
                  >
                    <div
                      class=""
                    >
                      <div
                        class="flex cursor-default select-none items-center p-1 text-darkText duration-200 dark:text-lightText"
                      >
                        <div
                          class="z-10 flex items-center space-x-2 group-hover:text-primaryColor dark:group-hover:text-primaryColor "
                        >
                          <p>
                            Support
                          </p>
                          <svg
                            aria-hidden="true"
                            class="svg-inline--fa fa-chevron-down duration-200"
                            data-icon="chevron-down"
                            data-prefix="fas"
                            focusable="false"
                            role="img"
                            viewBox="0 0 512 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                              fill="currentColor"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="col-span-1 flex items-center justify-end space-x-1 uppercase desktop:space-x-4  desktop:text-lg "
                >
                  <a
                    class="flex items-center space-x-1 rounded-lg px-3 py-1 hover:text-primaryColor dark:hover:text-primaryColor tablet:font-semibold desktop:space-x-2"
                    href="/login"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-user "
                      data-icon="user"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
                        fill="currentColor"
                      />
                    </svg>
                    <p>
                      Login
                    </p>
                  </a>
                  <div
                    class="rounded-lg bg-unhoveringBg hover:bg-hoveringBg"
                  >
                    <div
                      class=""
                    >
                      <div
                        class="flex border border-none px-1.5 py-1 desktop:px-2"
                      >
                        <div
                          class="flex items-center space-x-2 px-2 py-0.5 text-black"
                        >
                          <svg
                            aria-hidden="true"
                            class="svg-inline--fa fa-cart-shopping "
                            data-icon="cart-shopping"
                            data-prefix="fas"
                            focusable="false"
                            role="img"
                            viewBox="0 0 576 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M0 24C0 10.7 10.7 0 24 0H69.5c22 0 41.5 12.8 50.6 32h411c26.3 0 45.5 25 38.6 50.4l-41 152.3c-8.5 31.4-37 53.3-69.5 53.3H170.7l5.4 28.5c2.2 11.3 12.1 19.5 23.6 19.5H488c13.3 0 24 10.7 24 24s-10.7 24-24 24H199.7c-34.6 0-64.3-24.6-70.7-58.5L77.4 54.5c-.7-3.8-4-6.5-7.9-6.5H24C10.7 48 0 37.3 0 24zM128 464a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zm336-48a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
                              fill="currentColor"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="w-full py-10 tablet:py-12 desktop:py-16"
        >
          <div
            class="bg-lightBg py-2 dark:bg-darkBg desktopLarge:py-6"
          >
            <div
              class="container"
            >
              <div
                class="relative grid grid-cols-12 gap-4 desktop:gap-8 desktopLarge:gap-16"
              >
                <div
                  class="col-span-4"
                >
                  <div
                    class="sticky left-0 top-12 flex-col rounded-xl bg-lightColor700 p-6 text-darkText dark:bg-darkColor700 dark:text-lightText tablet:top-14 desktop:top-20"
                  >
                    <div
                      class="flex items-center justify-between"
                    >
                      <p
                        class="w-full text-xl font-medium desktop:text-2xl desktopLarge:text-3xl"
                      >
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse  dark:bg-white/10 css-7uzo9y-MuiSkeleton-root"
                          style="width: 60%;"
                        />
                      </p>
                    </div>
                    <div
                      class="mt-2"
                    >
                      <span
                        class="text-base font-medium desktop:text-lg desktopLarge:text-xl"
                      >
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse  dark:bg-white/10 css-7uzo9y-MuiSkeleton-root"
                          style="width: 20%;"
                        />
                      </span>
                    </div>
                    <div
                      class="mt-8 w-full rounded-lg border border-black/20 p-4 dark:border-white/20"
                    >
                      <div
                        class="flex items-center justify-between"
                      >
                        <p
                          class="w-full text-base font-medium desktop:text-lg desktopLarge:text-xl"
                        >
                          <span
                            class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse  dark:bg-white/10 css-7uzo9y-MuiSkeleton-root"
                            style="width: 20%;"
                          />
                        </p>
                        <p
                          class="text-sm text-darkText/60 dark:text-lightText/60 desktop:text-base "
                        >
                          <span
                            class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse  dark:bg-white/10 css-7uzo9y-MuiSkeleton-root"
                            style="width: 20%;"
                          />
                        </p>
                      </div>
                      <div
                        class="max-h-64 w-full overflow-auto py-4"
                      >
                        <div
                          class="grid w-full grid-cols-3 gap-4"
                        >
                          <div
                            class="col-span-1 rounded-xl"
                          >
                            <div
                              class="relative w-full pt-[100%]"
                            >
                              <span
                                class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse absolute left-0 top-0 dark:bg-white/10 css-7uzo9y-MuiSkeleton-root"
                                style="width: 100%; height: 100%;"
                              />
                            </div>
                          </div>
                          <div
                            class="col-span-1 rounded-xl"
                          >
                            <div
                              class="relative w-full pt-[100%]"
                            >
                              <span
                                class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse absolute left-0 top-0 dark:bg-white/10 css-7uzo9y-MuiSkeleton-root"
                                style="width: 100%; height: 100%;"
                              />
                            </div>
                          </div>
                          <div
                            class="col-span-1 rounded-xl"
                          >
                            <div
                              class="relative w-full pt-[100%]"
                            >
                              <span
                                class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse absolute left-0 top-0 dark:bg-white/10 css-7uzo9y-MuiSkeleton-root"
                                style="width: 100%; height: 100%;"
                              />
                            </div>
                          </div>
                          <div
                            class="col-span-1 rounded-xl"
                          >
                            <div
                              class="relative w-full pt-[100%]"
                            >
                              <span
                                class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse absolute left-0 top-0 dark:bg-white/10 css-7uzo9y-MuiSkeleton-root"
                                style="width: 100%; height: 100%;"
                              />
                            </div>
                          </div>
                          <div
                            class="col-span-1 rounded-xl"
                          >
                            <div
                              class="relative w-full pt-[100%]"
                            >
                              <span
                                class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse absolute left-0 top-0 dark:bg-white/10 css-7uzo9y-MuiSkeleton-root"
                                style="width: 100%; height: 100%;"
                              />
                            </div>
                          </div>
                          <div
                            class="col-span-1 rounded-xl"
                          >
                            <div
                              class="relative w-full pt-[100%]"
                            >
                              <span
                                class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse absolute left-0 top-0 dark:bg-white/10 css-7uzo9y-MuiSkeleton-root"
                                style="width: 100%; height: 100%;"
                              />
                            </div>
                          </div>
                          <div
                            class="col-span-1 rounded-xl"
                          >
                            <div
                              class="relative w-full pt-[100%]"
                            >
                              <span
                                class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse absolute left-0 top-0 dark:bg-white/10 css-7uzo9y-MuiSkeleton-root"
                                style="width: 100%; height: 100%;"
                              />
                            </div>
                          </div>
                          <div
                            class="col-span-1 rounded-xl"
                          >
                            <div
                              class="relative w-full pt-[100%]"
                            >
                              <span
                                class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse absolute left-0 top-0 dark:bg-white/10 css-7uzo9y-MuiSkeleton-root"
                                style="width: 100%; height: 100%;"
                              />
                            </div>
                          </div>
                          <div
                            class="col-span-1 rounded-xl"
                          >
                            <div
                              class="relative w-full pt-[100%]"
                            >
                              <span
                                class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse absolute left-0 top-0 dark:bg-white/10 css-7uzo9y-MuiSkeleton-root"
                                style="width: 100%; height: 100%;"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="w-full"
                    >
                      <div
                        class="mt-6 flex items-center justify-between text-xs desktop:text-sm desktopLarge:text-base"
                      >
                        <div
                          class="flex items-center space-x-2"
                        >
                          <p
                            class="text-darkText dark:text-lightText"
                          />
                          <span
                            class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse  dark:bg-white/10 css-7uzo9y-MuiSkeleton-root"
                            style="width: 20%;"
                          />
                        </div>
                        <div
                          class="flex items-center space-x-1 text-darkText/60 dark:text-lightText/60"
                        >
                          <span
                            class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse  dark:bg-white/10 css-7uzo9y-MuiSkeleton-root"
                            style="width: 20%;"
                          />
                        </div>
                      </div>
                      <div
                        class="mt-4 flex justify-between"
                      >
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse  dark:bg-white/10 css-7uzo9y-MuiSkeleton-root"
                          style="width: 20%;"
                        />
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse  dark:bg-white/10 css-7uzo9y-MuiSkeleton-root"
                          style="width: 20%;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="col-span-8"
                >
                  <div
                    class="relative"
                  >
                    <div
                      class="relative w-full cursor-zoom-in overflow-hidden rounded-xl border border-black/10 bg-lightColor700 pt-[75%] dark:border-white/10 dark:bg-darkColor700 "
                    >
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse absolute left-0 top-0 dark:bg-white/10 css-7uzo9y-MuiSkeleton-root"
                        style="width: 100%; height: 100%;"
                      />
                    </div>
                  </div>
                  <div
                    class="h-80"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="w-full bg-lightColor700 pb-12 pt-12 duration-200 dark:bg-darkColor700"
        >
          <div
            class="container"
          >
            <div
              class=" grid grid-cols-5 gap-1 text-sm font-medium uppercase text-black duration-200 dark:text-white desktop:gap-2 desktop:text-xl desktopLarge:gap-4 desktopLarge:text-2xl"
            >
              <div
                class="col-span-1 flex items-center justify-center"
              >
                follow us
              </div>
              <div
                class="col-span-1 flex items-center px-2 desktop:px-4 desktopLarge:px-8"
              >
                Hareta
              </div>
              <div
                class="col-span-1 flex items-center px-0 text-sm desktop:px-2 desktop:text-xl desktopLarge:text-2xl"
              >
                our main office
              </div>
              <div
                class="col-span-1 flex items-center px-6 desktop:px-8"
              >
                support
              </div>
              <div
                class="col-span-1 flex items-center px-2 desktop:px-4 "
              >
                contact
              </div>
            </div>
            <div
              class="mt-4 grid grid-cols-5 justify-center gap-1 text-xs text-darkText/80 duration-200 dark:text-lightText/80 desktop:gap-2 desktop:text-base desktopLarge:gap-4"
            >
              <div
                class="col-span-1"
              >
                <div
                  class="flex items-center justify-center space-x-1 text-lg desktop:space-x-2 desktop:text-xl desktopLarge:text-2xl"
                >
                  <a
                    href="https://www.facebook.com/HaretaWorkshop"
                    target="_blank"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-facebook cursor-pointer p-1 hover:text-primaryColor dark:hover:text-primaryColor"
                      data-icon="facebook"
                      data-prefix="fab"
                      focusable="false"
                      role="img"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
                        fill="currentColor"
                      />
                    </svg>
                  </a>
                  <a
                    href="https://www.instagram.com/hareta.workshop/"
                    target="_blank"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-instagram cursor-pointer p-1 hover:text-primaryColor dark:hover:text-primaryColor"
                      data-icon="instagram"
                      data-prefix="fab"
                      focusable="false"
                      role="img"
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                        fill="currentColor"
                      />
                    </svg>
                  </a>
                  <a
                    href="/"
                    target="_blank"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-discord cursor-pointer p-1 hover:text-primaryColor dark:hover:text-primaryColor"
                      data-icon="discord"
                      data-prefix="fab"
                      focusable="false"
                      role="img"
                      viewBox="0 0 640 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M524.531,69.836a1.5,1.5,0,0,0-.764-.7A485.065,485.065,0,0,0,404.081,32.03a1.816,1.816,0,0,0-1.923.91,337.461,337.461,0,0,0-14.9,30.6,447.848,447.848,0,0,0-134.426,0,309.541,309.541,0,0,0-15.135-30.6,1.89,1.89,0,0,0-1.924-.91A483.689,483.689,0,0,0,116.085,69.137a1.712,1.712,0,0,0-.788.676C39.068,183.651,18.186,294.69,28.43,404.354a2.016,2.016,0,0,0,.765,1.375A487.666,487.666,0,0,0,176.02,479.918a1.9,1.9,0,0,0,2.063-.676A348.2,348.2,0,0,0,208.12,430.4a1.86,1.86,0,0,0-1.019-2.588,321.173,321.173,0,0,1-45.868-21.853,1.885,1.885,0,0,1-.185-3.126c3.082-2.309,6.166-4.711,9.109-7.137a1.819,1.819,0,0,1,1.9-.256c96.229,43.917,200.41,43.917,295.5,0a1.812,1.812,0,0,1,1.924.233c2.944,2.426,6.027,4.851,9.132,7.16a1.884,1.884,0,0,1-.162,3.126,301.407,301.407,0,0,1-45.89,21.83,1.875,1.875,0,0,0-1,2.611,391.055,391.055,0,0,0,30.014,48.815,1.864,1.864,0,0,0,2.063.7A486.048,486.048,0,0,0,610.7,405.729a1.882,1.882,0,0,0,.765-1.352C623.729,277.594,590.933,167.465,524.531,69.836ZM222.491,337.58c-28.972,0-52.844-26.587-52.844-59.239S193.056,219.1,222.491,219.1c29.665,0,53.306,26.82,52.843,59.239C275.334,310.993,251.924,337.58,222.491,337.58Zm195.38,0c-28.971,0-52.843-26.587-52.843-59.239S388.437,219.1,417.871,219.1c29.667,0,53.307,26.82,52.844,59.239C470.715,310.993,447.538,337.58,417.871,337.58Z"
                        fill="currentColor"
                      />
                    </svg>
                  </a>
                  <a
                    href="/"
                    target="_blank"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-youtube cursor-pointer p-1 hover:text-primaryColor dark:hover:text-primaryColor"
                      data-icon="youtube"
                      data-prefix="fab"
                      focusable="false"
                      role="img"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
                        fill="currentColor"
                      />
                    </svg>
                  </a>
                </div>
              </div>
              <div
                class="col-span-1 flex flex-col space-y-2 px-2 desktop:px-4 desktopLarge:px-8"
              >
                <a
                  class="hover:text-primaryColor dark:hover:text-primaryColor"
                  href="/hareta.me"
                >
                  Hareta workshop
                </a>
              </div>
              <p
                class="col-span-1 text-pretty px-0 desktop:px-2"
              >
                Bcons Bee Building, Binh An, Di An, Binh Duong, Viet Nam
              </p>
              <div
                class="col-span-1 flex flex-col space-y-2 px-6 desktop:px-8"
              >
                <a
                  class="text-pretty"
                  href="/privacy-and-terms"
                >
                  Privacy & Terms
                </a>
                <a
                  class="text-pretty"
                  href="/order-tracking"
                >
                  Order Tracking
                </a>
              </div>
              <div
                class="col-span-1 flex flex-col justify-start space-y-2 px-2 desktop:px-4"
              >
                <p>
                  contact.hareta@gmail.com
                </p>
                <p>
                  0818266375
                </p>
              </div>
            </div>
            <div
              class="relative grid grid-cols-5 gap-1 desktop:gap-2 desktopLarge:gap-4"
            >
              <div
                class="col-span-2 flex items-center justify-center space-x-2 desktop:space-x-4"
              >
                <div
                  class="relative h-32 w-32 desktop:h-40 desktop:w-40"
                >
                  <img
                    alt="Hareta"
                    class="absolute left-0 top-0 h-full w-full object-scale-down"
                    src="/images/haretaWorkshopLogo.png"
                  />
                </div>
                <div
                  class="uppercase text-darkText/80 duration-200 dark:text-lightText/80"
                >
                  <p
                    class="text-lg desktop:text-xl desktopLarge:text-2xl"
                  >
                    hareta
                  </p>
                  <p
                    class="pl-4  text-sm desktop:text-base desktopLarge:text-lg"
                  >
                    workshop
                  </p>
                </div>
              </div>
              <div
                class="absolute bottom-0 flex w-full items-center justify-center text-xs text-darkText/60 duration-200 dark:text-lightText/60 desktop:text-sm"
              >
                Copyright © 2023 Hareta Workshop. All rights reserved
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="Toastify"
      >
        <div
          class="Toastify__toast-container Toastify__toast-container--top-right"
        >
          <div
            class="Toastify__toast Toastify__toast-theme--light Toastify__toast--error Toastify__toast--close-on-click Toastify--animate Toastify__bounce-enter--top-right"
            id="1"
            style="--nth: 1; --len: 1;"
          >
            <div
              class="Toastify__toast-body"
              role="alert"
            >
              <div
                class="Toastify__toast-icon Toastify--animate-icon Toastify__zoom-enter"
              >
                <svg
                  fill="var(--toastify-icon-color-error)"
                  height="100%"
                  viewBox="0 0 24 24"
                  width="100%"
                >
                  <path
                    d="M11.983 0a12.206 12.206 0 00-8.51 3.653A11.8 11.8 0 000 12.207 11.779 11.779 0 0011.8 24h.214A12.111 12.111 0 0024 11.791 11.766 11.766 0 0011.983 0zM10.5 16.542a1.476 1.476 0 011.449-1.53h.027a1.527 1.527 0 011.523 1.47 1.475 1.475 0 01-1.449 1.53h-.027a1.529 1.529 0 01-1.523-1.47zM11 12.5v-6a1 1 0 012 0v6a1 1 0 11-2 0z"
                  />
                </svg>
              </div>
              <div>
                Network Error
              </div>
            </div>
            <button
              aria-label="close"
              class="Toastify__close-button Toastify__close-button--light"
              type="button"
            >
              <svg
                aria-hidden="true"
                viewBox="0 0 14 16"
              >
                <path
                  d="M7.71 8.23l3.75 3.75-1.48 1.48-3.75-3.75-3.75 3.75L1 11.98l3.75-3.75L1 4.48 2.48 3l3.75 3.75L9.98 3l1.48 1.48-3.75 3.75z"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
            <div
              aria-hidden="false"
              aria-label="notification timer"
              class="Toastify__progress-bar Toastify__progress-bar--animated Toastify__progress-bar-theme--light Toastify__progress-bar--error"
              role="progressbar"
              style="animation-duration: 5000ms; animation-play-state: paused; opacity: 1;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
